### ISSUE TITLE
Hidden queued job interrupts to update the event status.

### ISSUE THREAD

**Issue by yoshwata** [screwdriver-cd/screwdriver#2004]
**What happened**:
It is rare case but it is what actually happened. 

A build `11111929` and another build `11111926` was generated and these has same Job and Event id.

<img width="802" alt="duplicate-build" src="https://user-images.githubusercontent.com/32473622/75864708-c7704700-5e45-11ea-9538-b06ce79e12b9.png">

In this situation the event `6286147` is `QUEUED` status permanently, so it seemes like that the event works continuingly even though all build has done. The reason is `11111926` is still on setup step.

It is impossible to access `11111926` build screen from UI, in other words the link to build is just for `11111929`. Accessing `HOST/builds/11111929` directly is possible.

**What you expected to happen**:
Duplicate builds which has same Event and Job id is not created, or deleted automatically if it was regenerated.

**How to reproduce it**:
I could not reproduce it.

**Comment by jithine** [comment]
@yoshwata We've seen this situation. In fact @pritamstyz4ever has added a check to delete them if blocked by catches it. But it's still happening and we've seen them in `RUNNING` state as well.

**Comment by jithine** [comment]
This is same as https://github.com/screwdriver-cd/screwdriver/issues/2071 closing in favor of it. Will track fix there
